utils/wasm-instrument/src/lib.rs

```diff
@@ -213,17 +204,27 @@ pub fn inject<R: Rules>(
     ];
 
     // determine cost for successful execution
+    let mut block_of_code = false;
+
     let cost_blocks = match elements
         .iter()
-        .take(7)
-        // block with update instructions
-        .chain(elements.iter().skip(10).take(7))
+        .filter(|instruction| match instruction {
+            Instruction::If(_) => {
+                block_of_code = true;
+                true
+            }
+            Instruction::End => {
+                block_of_code = false;
+                false
+            }
+            _ => !block_of_code,
+        })
```

Comment:
Sorry cannot agree with the edits. This cost calculation is hardwired to the function body so if it would be changed sometime then these lines should be adjusted accordingly the same way as with "magic constants".

Reply:
> Sorry cannot agree with the edits. This cost calculation is hardwired to the function body so if it would be changed sometime then these lines should be adjusted accordingly the same way as with "magic constants".

Why? It's more general than some previous skips etc.